src/v0/sources/mailjet/transform.js

```diff
@@ -7,45 +7,49 @@ const Message = require('../message');
 // import mapping json using JSON.parse to preserve object key order
 const mapping = JSON.parse(fs.readFileSync(path.resolve(__dirname, './mapping.json'), 'utf-8'));
 
-function process(event) {
-  const message = new Message(`MailJet`);
-
-  // event type is always track
-  const eventType = 'track';
-
-  message.setEventType(eventType);
-
-  message.setEventName(event.event);
-
-  message.setPropertiesV2(event, mapping);
-
-  if (event.time) {
-    const ts = new Date(event.time * 1000).toISOString();
-    message.setProperty('originalTimestamp', ts);
-  }
-
-  const externalId = [];
-  // setting up mailjet contact_id and list_id to externalId
-  if (event.mj_contact_id) {
-    externalId.push({
-      type: 'mailjetContactId',
-      id: event.mj_contact_id,
-    });
-  }
-  if (event.mj_list_id) {
-    externalId.push({
-      type: 'mailjetListId',
-      id: event.mj_list_id,
-    });
-  }
-  message.context.externalId = externalId;
-
-  if (!message.userId && event.email) {
-    // Treating userId as unique identifier
-    // If userId is not present, then generating it from email using md5 hash function
-    message.userId = md5(event.email);
-  }
-  return message;
-}
+const processEvent = (eventList) => {
+  const messages = [];
+  eventList.forEach((event) => {
+    const message = new Message(`MailJet`);
+    // event type is always track
+    const eventType = 'track';
+    message.setEventType(eventType);
+    message.setEventName(event.event);
+    message.setPropertiesV2(event, mapping);
+
+    if (event.time) {
+      const ts = new Date(event.time * 1000).toISOString();
+      message.setProperty('originalTimestamp', ts);
+    }
+
+    const externalId = [];
+    // setting up mailjet contact_id and list_id to externalId
+    if (event.mj_contact_id) {
+      externalId.push({
+        type: 'mailjetContactId',
+        id: event.mj_contact_id,
+      });
+    }
+    if (event.mj_list_id) {
+      externalId.push({
+        type: 'mailjetListId',
+        id: event.mj_list_id,
+      });
+    }
+    message.context.externalId = externalId;
+
+    if (!message.userId && event.email) {
+      // Treating userId as unique identifier
+      // If userId is not present, then generating it from email using md5 hash function
+      message.userId = md5(event.email);
+    }
+    messages.push(message);
+  });
+
+  return messages;
+};
+
+// This fucntion just converts the incoming payload to array of already not and sends it to processEvent
+const process = (event) => (Array.isArray(event) ? processEvent(event) : processEvent([event]));
```

Comment:
we could achieve the same thing with more clarity:

my converting original process function as processEvent which only process single event.
// using https://github.com/rudderlabs/rudder-workflow-engine/blob/bc179fb899892711e83e5e326dd69b2fd6f8e10c/src/bindings/common.ts#L30
const process = (events) => {
  const eventsArray = toArray(events);
  return eventsArray.map(process);
}

Reply:
Done